### DISCUSSION TITLE
ENV vars not being applied

### DISCUSSION THREAD

**Discussion by mbajur** [basecamp/kamal#826] [General]
Hey! I have a kamal-based app with a following `deploy.yml` file:

https://github.com/mbajur/activity_pub_app/blob/39709bc764f99d1074b335e39dc5d9ba983ca4e4/config/deploy.yml

However, no matter what i do, the `clear` env vars are not being applied to my production containers. Only the `secret` ones does. I tried running `kamal env push`, `kamal traefik reboot`, deployed app several times but in order to make my app work, i had to temporarily hardcode these variables beccause none of them is available in my container. Neither in `ENV` nor as a `printenv` output. I also have another app running on kamal and never experienced such issue in there.

The broken kamal app is on 1.5.2 but i have it installed globally so the working one is on the same version :)

Any ideas what am i missing in here? Thanks!

**Comment by nuschk** [comment]
Yes, I have the same problem I think: #803 . My solution was to drop down to 1.4 for the time being, I sure hope it gets fixed as I think it's broken this way.

I don't know, maybe most people have a vault setup in their environments and thus don't see this problem...

**Comment by nickhammond** [comment]
@mbajur what does the `docker run` line look like in the kamal deploy output? Does it utilize `--env-file` and you're not seeing any `--env` with it?
